### ISSUE TITLE
I can't understand the process of pretraining generator

### ISSUE THREAD

**Issue by keruoya** [LS4GAN/uvcgan2#46]
Thank you for your excellent work ! After checking your code for pretraining generator , I still feel confused about it , could you give me more insights of it?

**Comment by usert5432** [comment]
Hi @keruoya

Thanks! I will be happy to help clarify the pretraining generator code.

I was wondering, if you would be able to suggest some specific parts that are confusing? That may help me to give more targeted suggestions.

**Comment by keruoya** [comment]
> Hi [@keruoya](https://github.com/keruoya)
> 
> Thanks! I will be happy to help clarify the pretraining generator code.
> 
> I was wondering, if you would be able to suggest some specific parts that are confusing? That may help me to give more targeted suggestions.

Thank you for your reply ! You mentioned that the process of pretraining generator is like the way of Bert , I wonder that is this this operation implemented on the target domain?

**Comment by keruoya** [comment]
> Hi [@keruoya](https://github.com/keruoya)
> 
> Thanks! I will be happy to help clarify the pretraining generator code.
> 
> I was wondering, if you would be able to suggest some specific parts that are confusing? That may help me to give more targeted suggestions.

Sorry to bother you , I have struggled for this question so long , and I also want to know the reason of pretraining generator like this way . Thank you very much !

**Comment by usert5432** [comment]
Hi @keruoya, Thank you for the question. Let me elaborate on how we setup our pre-training:

1. We take input images (256×256 pixels)
2. Split into a grid of tiles (32×32 pixels each)
3. Randomly mask 40% of tiles independently
4. Train a generator to reconstruct the masked tiles (input: masked image, target: original unmasked image)

We pretrain the generator on both domains simultaneously (not only target domain). The main reason we use this approach is because we found empirically that it gives better results than training networks from scratch.

Please let me know if I should elaborate further on any part.
